src/client/Inode.cc

```diff
@@ -197,6 +197,20 @@ bool Inode::caps_issued_mask(unsigned mask)
     touch_cap(auth_cap);
     return true;
   }
+  // It should fall in here, since this MDS should be the authority for that directory...
+  // I think it isn't falling in here because it thinks its looking for a file instead of 
+  // a directory - so when it checks the auth_cap->issued & mask, it fails
+  if (is_dir()) {
+    unsigned int a = 1;
+    unsigned int newmask = mask & (~(a << CEPH_CAP_SFILE));
+    if (auth_cap &&
+        cap_is_valid(auth_cap) &&
+        (auth_cap->issued & newmask) == newmask) {
+      touch_cap(auth_cap);
+      return true;
+    }
+  }
+
```

Comment:
I was seeing an excessive amount of getattr()s from the client. This seemed to fix it - but as you can tell from the comment I'm incredibly unsure. I'll get rid of it and commit a separate bugfix.